### DISCUSSION TITLE
Ideas / To Do / Work In Progress

### DISCUSSION THREAD

**Discussion by 63BeetleSmurf** [63BeetleSmurf/PinnyNotes#3] [Ideas]
This is a dumping ground for my ideas so I don't forget them.

Feel free to post your own, but please mark if you are working on something and add a link to your fork encase someone wants to help out.

### Reminders
Set a timer to have a note brought to your attention at a specific time or after a specified period. Snooze and overdue reminders could be part of this to.

### Groups
Multiple groups that can be shown at the same time or not. User should pick a group colour and this will be used as the border or title colour on all notes from that group.
Could allow creating shortcuts on the desktop which would bring up a particular note group.

### Search Notes in Management Window
A search box on top of the window would be handy to quickly search for notes

### Nextcloud Integration
Would require the Management Window / Save Notes to be implemented.

### Date stamp tools
Initial tool done, just need to decided what formats to offer in the menu.
Tools to reformat a date.

### Search & Replace
Search & Replace with regular expressions (e.g. as found in [Geany](https://www.geany.org/))

### Trim Selected
Allow trimming selected text from lines. May be confusing though and better if search & replace was available.

### Multi Language
May need to use another spell checker like Hunspell library and deal with right-to-left text.

### Customisable Themes
Allow adding, editing and removing themes.

### Quick Settings Menu
Add a menu items to access common editor settings from the context menu such as spell check or word-wrap mentioned above.

**Comment by ghost** [comment]
I'm literally looking everywhere for an open source sticky note for Windows; I think I've probably tried them all and they all lack the most important thing: the reminder. A simple alert, maybe with a window that flashes at the chosen day and time.

Your sticky note is really great and, in my opinion, with the addition of a reminder it would be perfect. So I agree with your idea of ​​adding a reminder, and I hope it can be the next addition!

And thanks for your work, it is really appreciated!

**Reply by 63BeetleSmurf** [reply]
Thanks, glad it's been of use to you.

Reminders is something I want as well and would probably use daily, but the management window needs added first otherwise it'll be messy I think. Setting a reminder is simple enough from a single note but getting into snoozing or seeing pending reminders needs more centralised management.

I'm in the process of moving from a somewhat MVVM architecture to MVP, which I think will suit this project better long term.

I'm hoping to have a version 2 out around new year, and of reminders isn't originally included it will be the next thing added.

Thanks for the feedback.

**Reply by ghost** [reply]
Well then, I will patiently wait for the new releases with the reminder to try them! 😎🍻

**Reply by PeterPan8** [reply]
My exact thought! I would very likely use that app if it had a reminder system, a bit like PNotes. But for now, I'll pass....

**Reply by 63BeetleSmurf** [reply]
Now we finally have the management window I'll be looking at adding reminders next.
